chapters/annotations.tex

```diff
@@ -8,6 +8,21 @@ \chapter{Annotations}\label{annotations}
 (For replaceable class declarations with a \lstinline[language=grammar]!constraining-clause! also refer to \cref{constraining-clause-annotations}.)
 The specification in this document defines the semantic meaning if a tool implements any of these annotations.
 
+\section{Notation for Annotation Definitions}\label{notation-for-annotation-definitions}
+
+Annotations are defined using the syntactic forms of Modelica component declarations and record definitions, with the following special semantics.
+If the annotation is described by a component declaration, the annotation is used in the form of a value modifier for the same name.
+If the annotation is described by a \lstinline!record! \emph{class} the annotation is used in the form of a modifier for a \lstinline!record! \emph{component} with the same name.
+
+A declaration equation for a component or record member specifies a default to be used when no corresponding annotation is given.
+Default behavior can also be specified in the text as an alternative to a declaration equation.
+Component declaration annotations always have default behavior, as all annotations are optional.
+When there is no declaration equation and no other explanation of default behavior in the text for a record member, an annotation modifier for the record must contain a modifier for that member.
+
+When an annotation is defined with a component variability prefix (\cref{component-variability-prefixes-discrete-parameter-constant}), this restricts the allowed variability of corresponding annotation modifiers analogously to the rules in \cref{variability-of-expressions}.
+If the annotation is declared as an \lstinline!/*evaluable*/ parameter! the corresponding modifier is further restricted to be evaluable.
+If the annotation is declared as a \lstinline!/*literal*/ constant! the corresponding modifier is further restricted to be a literal value.
+
```

Comment:
Until we have a viable alternative to `{}` we should define what it means so that we can use it without not conforming to our own notation.  Disallowing this as part of this PR would be a major scope creep in my opinion.

The day we find an alternative, there will be no backwards compatibility issue of changing the notation, as it is only used within the specification itself.
```suggestion
For an array record member, the special expression \lstinline!{}! in a declaration equation indicates that the array is empty by default.
```

Reply:
I have broken that out into another PR.

Reply:
I'm resolving this in order to start over with a new suggestion originating from that PR.